core/src/tangram.cpp

```diff
@@ -442,82 +445,28 @@ void Map::render() {
         for (const auto& style : impl->scene->styles()) {
             style->onBeginDrawSelectionFrame(impl->renderState, impl->view, *(impl->scene));
 
-            for (const auto& tile : impl->tileManager.getVisibleTiles()) {
-                style->drawSelectionFrame(impl->renderState, *tile);
-            }
-        }
-
-        // Resolve feature selection queries
-        for (const auto& selectionQuery : impl->featureSelectionQueries) {
-
-            float x = selectionQuery.position.x / impl->view.getWidth();
-            float y = (1.f - (selectionQuery.position.y / impl->view.getHeight()));
-
-            // TODO: read with a scalable thumb size
-            GLuint color = impl->selectionBuffer->readAt(x, y);
-
-            auto position = std::array<float, 2>{{selectionQuery.position.x,
-                                                  selectionQuery.position.y}};
-            bool found = false;
-            if (color != 0) {
+            if (hasGeometryFeatureQuery) {
                 for (const auto& tile : impl->tileManager.getVisibleTiles()) {
-                    if (auto props = tile->getSelectionFeature(color)) {
-
-                        FeaturePickResult queryResult(props, position);
-                        selectionQuery.onFeatureSelection(&queryResult);
-                        found = true;
-                        break;
-                    }
+                    style->drawSelectionFrame(impl->renderState, *tile);
                 }
             }
-            if (!found) {
-                selectionQuery.onFeatureSelection(nullptr);
-                continue;
-            }
-        }
-
-        impl->featureSelectionQueries.clear();
-
-        // Resolve label selection queries
-        for (const auto& labelQuery : impl->labelSelectionQueries) {
-
-            float x = labelQuery.position.x / impl->view.getWidth();
-            float y = (1.f - (labelQuery.position.y / impl->view.getHeight()));
-
-            // TODO: read with a scalable thumb size and iterate over the read colors
-            GLuint color = impl->selectionBuffer->readAt(x, y);
 
-            // Retrieve the label for this selection color
-            if (color == 0) {
-                labelQuery.onLabelSelection(nullptr);
-                continue;
-            }
-            auto label = impl->labels.getLabel(color);
-            if (!label.first) {
-                labelQuery.onLabelSelection(nullptr);
-                continue;
-            }
-            auto props = label.second->getSelectionFeature(label.first->options().featureId);
-            if (!props) {
-                labelQuery.onLabelSelection(nullptr);
-                continue;
+            if (hasMarkerQuery) {
```

Comment:
This is an interesting and substantial design decision. If a client makes a query for a feature or label (but not a marker) on a point that is covered by a marker, the result can be a feature or label that is _under_ the marker. This is potentially useful, but diverges from the WYSIWYG behavior of this feature.

I'm not sure yet whether this is good or bad.